server/datastore/mysql/android.go

```diff
@@ -660,3 +660,235 @@ func (ds *Datastore) NewAndroidPolicyRequest(ctx context.Context, req *fleet.MDM
+	-- by design, "include" labels cannot match if they are broken (the host cannot be
+	-- a member of a deleted label).
+	SELECT
+		macp.profile_uuid,
+		macp.name,
+		h.uuid as host_uuid,
+		h.id as host_id,
+		COUNT(*) as count_profile_labels,
+		COUNT(mcpl.label_id) as count_non_broken_labels,
+		COUNT(lm.label_id) as count_host_labels,
+		0 as count_host_updated_after_labels
+	FROM
+		mdm_android_configuration_profiles macp
+			JOIN hosts h
+				ON h.team_id = macp.team_id OR (h.team_id IS NULL AND macp.team_id = 0)
+			JOIN android_devices ad
+				ON ad.host_id = h.id
+			JOIN mdm_configuration_profile_labels mcpl
+				ON mcpl.android_profile_uuid = macp.profile_uuid AND mcpl.exclude = 0 AND mcpl.require_all = 0
+			LEFT OUTER JOIN label_membership lm
+				ON lm.label_id = mcpl.label_id AND lm.host_id = h.id
+	WHERE
+		h.platform = 'android' AND
+		( %s )
+	GROUP BY
+		macp.profile_uuid, macp.name, h.uuid, h.id
+	HAVING
+		count_profile_labels > 0 AND count_host_labels >= 1
+`
+
+// ListMDMAndroidProfilesToSend is the android platform equivalent to
+// ListMDMAppleProfilesToInstall/Remove and
+// ListMDMWindowsProfilesToInstall/Remove. It plays a similar role but is quite
+// different in implementation since Android profiles are fundamentally
+// different from those platforms - the "configuration profiles" are just
+// fragments of the full (and unique per host) "Android policy" to apply to a
+// host, so as soon as there is a change to the set of profiles to apply to a
+// host, the full list of applicable profiles are needed to generate the
+// resulting policy.
+//
+// That is, profiles are not applied individually, but as a merged set. For
+// that reason, there is no "to install" and "to remove", as removing a profile
+// is just not including it in the merged set of profiles to apply.
+//
+// So with that in mind, what this method does is return the full set of
+// applicable profiles for each host that has a change in that set of
+// applicable profiles, so that it needs to be sent again.
+//
+// See https://github.com/fleetdm/fleet/issues/32032#issuecomment-3229548389
+// for more details on the rationale of that approach.
+func (ds *Datastore) ListMDMAndroidProfilesToSend(ctx context.Context) ([]*fleet.MDMAndroidProfilePayload, error) {
+	var result []*fleet.MDMAndroidProfilePayload
+	err := ds.withTx(ctx, func(tx sqlx.ExtContext) error {
+		hostsWithChangesStmt := fmt.Sprintf(`
+	WITH ds AS ( %s )
+
+	SELECT
+		DISTINCT ds.host_uuid
+	FROM ds
+		INNER JOIN android_devices ad
+			ON ad.host_id = ds.host_id
+		LEFT OUTER JOIN host_mdm_android_profiles hmap
+			ON hmap.host_uuid = ds.host_uuid AND hmap.profile_uuid = ds.profile_uuid
+	WHERE
+	  -- at least one profile is missing from host_mdm_android_profiles
+		hmap.host_uuid IS NULL OR
+		-- profile was never sent or was updated after sent
+		-- TODO(ap): need to make sure we set it to NULL when profile is updated
+		hmap.included_in_policy_version IS NULL OR
```

Comment:
Assuming the gitops work will merge before this PR, I'll update the UPDATE profile logic to NULL this field (profiles can only ever be updated via gitops, the UI only supports INSERT/DELETE).